third_party/2and3/pytz/__init__.pyi

```diff
@@ -1,41 +1,40 @@
-# Stubs for pytz (Python 3.5)
-
+from typing import Optional, List, Set, Mapping, Union, overload
 import datetime
-from typing import Optional, List, Set, Dict, Union
-
-all_timezones = ...  # type: List
-all_timezones_set = ...  # type: Set
-common_timezones = ...  # type: List
-common_timezones_set = ...  # type: Set
-country_timezones = ...  # type: Dict
-country_names = ...  # type: Dict
 
+class BaseTzInfo(datetime.tzinfo):
+    zone: str = ...
+    def localize(self, dt: datetime.datetime, is_dst: bool = ...) -> datetime.datetime: ...
+    def normalize(self, dt: datetime.datetime) -> datetime.datetime: ...
 
-class _UTCclass(datetime.tzinfo):
-    zone = ...  # type: str
-    def fromutc(self, dt: datetime.datetime) -> datetime.datetime: ...
-    def utcoffset(self, dt: Optional[datetime.datetime]) -> datetime.timedelta: ...
+class _UTCclass(BaseTzInfo):
     def tzname(self, dt: Optional[datetime.datetime]) -> str: ...
+    def utcoffset(self, dt: Optional[datetime.datetime]) -> datetime.timedelta: ...
     def dst(self, dt: Optional[datetime.datetime]) -> datetime.timedelta: ...
-    def localize(self, dt: datetime.datetime, is_dst: bool = ...) -> datetime.datetime: ...
-    def normalize(self, dt: datetime.datetime, is_dst: bool = ...) -> datetime.datetime: ...
+
+class _StaticTzInfo(BaseTzInfo):
+    def tzname(self, dt: Optional[datetime.datetime], is_dst: Optional[bool] = ...) -> str: ...
+    def utcoffset(self, dt: Optional[datetime.datetime], is_dst: Optional[bool] = ...) -> datetime.timedelta: ...
+    def dst(self, dt: Optional[datetime.datetime], is_dst: Optional[bool] = ...) -> datetime.timedelta: ...
+
+class _DstTzInfo(BaseTzInfo):
+    def tzname(self, dt: Optional[datetime.datetime], is_dst: Optional[bool] = ...) -> str: ...
+    def utcoffset(self, dt: Optional[datetime.datetime], is_dst: Optional[bool] = ...) -> Optional[datetime.timedelta]: ...
+    def dst(self, dt: Optional[datetime.datetime], is_dst: Optional[bool] = ...) -> Optional[datetime.timedelta]: ...
+
+class UnknownTimeZoneError(KeyError): ...
+class InvalidTimeError(Exception): ...
+class AmbiguousTimeError(InvalidTimeError): ...
+class NonExistentTimeError(InvalidTimeError): ...
 
 utc: _UTCclass
 UTC: _UTCclass
+def timezone(zone: str) -> Union[_UTCclass, _StaticTzInfo, _DstTzInfo]: ...
```

Comment:
The docstring claims this returns a `datetime.tzinfo`. Might that be a better annotation? We try to avoid Unions in return types because they force users to do `isinstance` checks before using a value.

Reply:
Makes sense, the Union is not problematic here because the three Union members have almost the same interface.